mat64/cholesky.go

```diff
@@ -7,84 +7,180 @@ package mat64
 
 import (
 	"math"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+	"github.com/gonum/internal/asm"
 )
 
-type CholeskyFactor struct {
-	L   *Dense
-	SPD bool
-}
+// Cholesky calculates the Cholesky decomposition of the matrix A and returns
+// whether the matrix is positive definite. The returned matrix is either a
+// lower triangular matrix such that A = L * L^T or an upper triangular matrix
+// such that A = U^T * U depending on the upper parameter.
```

Comment:
I think there should be a "panics if triangular is of non-zero size and orientation does not match upper"